compiler/src/iree/compiler/Codegen/SPIRV/KernelConfig.cpp

```diff
@@ -158,33 +158,74 @@ namespace detail {
 LogicalResult setConvOpConfig(linalg::LinalgOp linalgOp,
                               const int64_t subgroupSize,
                               const int64_t bestTilingFactor) {
+  assert(isa<linalg::ConvolutionOpInterface>(*linalgOp));
   LLVM_DEBUG(llvm::dbgs() << "trying to deduce config as convolution...\n");
-  Type inputType = linalgOp.getDpsInputOperand(0)->get().getType();
-  ArrayRef<int64_t> inputShape = inputType.cast<ShapedType>().getShape();
+
+  if (isa<linalg::PoolingNhwcSumOp, linalg::PoolingNhwcMaxOp,
+          linalg::PoolingNhwcMaxUnsignedOp, linalg::PoolingNhwcMinOp,
+          linalg::PoolingNhwcMinUnsignedOp>(*linalgOp)) {
+    // TODO(antiagainst): Enable configuration for pooling op.
+    return failure();
+  }
+
   Type outputType = linalgOp.getDpsInitOperand(0)->get().getType();
   ArrayRef<int64_t> outputShape = outputType.cast<ShapedType>().getShape();
+  if (outputShape.size() != 4) return failure();  // Only 2-D conv for now
```

Comment:
(This doesn't look to be strictly NFC, there are some convolution variants in Linalg, e.g. [depthwise_conv_1d_nwc_wcm](https://github.com/llvm/llvm-project/blob/039f0e69a45ea007f1454ba1d4eb4628e3d20e28/mlir/include/mlir/Dialect/Linalg/IR/LinalgNamedStructuredOps.yaml#L2369) which I think will be able to get here and fail the assert below)